bazel/upload_systest_dep.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+#
+# Uploads a dependency to shared storage and returns the download URL.
+#
+# The path to the dependency should be specified as the first (and only) argument.
+#
+# The download URL is printed to stdout.
+
+# NOTE: This script uses bazel-remote as the CAS storage (implementation detail).
+
+# Look up a CAS key (provided as $1) through the redirect server.
+# If the key exists, then then download URL is returned (through stdout).
+# If the key does not exist, the empty string is returned.
+lookup_dep_url() {
+    REDIRECT_SERVER_URL="https://artifacts.idx.dfinity.network"
+    local redirect_url="$REDIRECT_SERVER_URL/cas/$1"
+    local result
+    result=$(curl --silent --head \
+        -w '%{http_code} %{redirect_url}' \
+        "$redirect_url" \
+        | tail -n1)
+
+    local result_code
+    result_code=$(cut -d' ' -f1 <<<"$result")
+    if [ "$result_code" == "404" ]; then
+        # The key was not found
+        return
+    fi
+
+    if [ "$result_code" != "307" ]; then
+        echo "Expected 404 or 307 when looking up dependency '$1', got '$result_code'" >&2
+        exit 1
+    fi
+
+    local result_url
+    result_url=$(cut -d' ' -f2 <<<"$result")
+    if [ -z "$result_url" ]; then
+        echo "Looking up dependency '$1' did not return a URL, got: '$result'" >&2
+        exit 1
+    fi
+
+    echo "$result_url"
+}
+
+dep_filename="${1:?Dependency not specified}"
+
+echo "Found dep to upload $dep_filename ($dep_sha256)" >&2
+result_url=$(lookup_dep_url "$dep_sha256")
+
+# First, figure out _if_ the dep should be uploaded (no point re-uploading several GBs
+# if it's been uploaded already)
+if [ -n "$result_url" ]; then
+    echo "dep '$dep_filename': already uploaded" >&2
+else
+    echo "dep '$dep_filename': not uploaded yet, uploading to $dep_upload_url" >&2
+
+    # We use bazel-remote as a CAS storage
+    UPLOAD_URL="http://server.bazel-remote.svc.cluster.local:8080/cas"
+    echo "Using upload URL: '$UPLOAD_URL'" >&2
+
+    # Upload the dep
+    dep_sha256=$(sha256sum "$dep_filename" | cut -d' ' -f1)
+    dep_upload_url="$UPLOAD_URL/$dep_sha256"
+    curl --silent --fail "$dep_upload_url" --upload-file "$dep_filename"
```

Comment:
I'm a bit concerned about thundering hurt scenarios where many system-tests start at exactly the same time and start uploading the same image. This shouldn't be too big of an issue if the upload time is small. What about we log the upload time here to get a good sense of this:
```suggestion
    curl --silent --fail "$dep_upload_url" --upload-file "$dep_filename" -w 'Total time: %{time_total}s'
```
You might need a `\n` in the format string.